### PR TITLE
T: Add tests for updating CrateDefMap in new resolve

### PR DESCRIPTION
**Pull request by dima74**
Add tests for updating `CrateDefMap` in [new resolve](https://github.com/intellij-rust/intellij-rust/issues/6217):

* `RsDefMapUpdateChangeSingleFileTest` - tests whether or not `CrateDefMap` should be updated after file modification
* `RsDefMapUpdateCreateNewFileTest` - tests whether or not `CrateDefMap` should be updated after creation of new file
* `RsMultipleDefMapsUpdateTest` - tests which exactly `CrateDefMap`s are updated when we modify some crate and then run resolve in some other crate

### CODE REVIEW

File: src/test/kotlin/org/rust/lang/core/resolve2/RsDefMapUpdateChangeSingleFileTest.kt [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+        pub extern crate foo;
+    """)
+
+    fun `test change mod item to mod decl`() = doTestChanged("""
+        mod foo {}
+    """, """
+        mod foo;
+    """)
+
+    fun `test change mod decl to mod item`() = doTestChanged("""
```
Comment (vlad20012) [inline src/test/kotlin/org/rust/lang/core/resolve2/RsDefMapUpdateChangeSingleFileTest.kt:216]: Hmm, looks like we often do suck a reverse test. Idea: we can teach `doTestChanged` to do a reverse test automatically! e.g.
```
InlineFile(before)
doTest(replaceFileContent(after), shouldChange)
doTest(replaceFileContent(before), shouldChange)
```

File: src/test/kotlin/org/rust/lang/core/resolve2/RsMultipleDefMapsUpdateTest.kt [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+        val getDefMapStamps = {
+            crateIds.associateWith {
+                val holder = project.defMapService.getDefMapHolder(it)
+                holder.defMap!!.timestamp
+            }
+        }
+        val defMapStampsOld = getDefMapStamps()
+
+        NonBlockingReadActionImpl.waitForAsyncTaskCompletion()
```
Comment (vlad20012) [inline src/test/kotlin/org/rust/lang/core/resolve2/RsMultipleDefMapsUpdateTest.kt:57]: Also works without these awaits

**Review by vlad20012 — APPROVED** [review]
LGTM, cool tests!

File: src/test/kotlin/org/rust/lang/core/resolve2/RsDefMapUpdateChangeSingleFileTest.kt [hunk 3]
```diff
@@ -0,0 +1,411 @@
...
+import org.intellij.lang.annotations.Language
+import org.rust.ExpandMacros
+import org.rust.UseNewResolve
+
+/** Tests whether or not [CrateDefMap] should be updated after file modification */
+@UseNewResolve
+@ExpandMacros  // needed to enable precise modification tracker
+class RsDefMapUpdateChangeSingleFileTest : RsDefMapUpdateTestBase() {
+
+    private fun type(text: String = "a"): () -> Unit = {
```
[the remark this answers is not in the document]
Comment (dima74) [reply src/test/kotlin/org/rust/lang/core/resolve2/RsDefMapUpdateChangeSingleFileTest.kt:20]: Thanks, I didn't know this. Does it apply only to utility methods, or also for test methods (`doTest`, `doTestChanged`, etc)?

[the remark this answers is not in the document]
Comment (vlad20012) [reply src/test/kotlin/org/rust/lang/core/resolve2/RsDefMapUpdateChangeSingleFileTest.kt]: For test methods too

**Comment by dima74** [comment]
bors r=vlad20012

**Pull request merged**